offsets_db_api/routers/projects.py

```diff
@@ -19,14 +19,38 @@
     ProjectType,
     ProjectWithClips,
 )
-from offsets_db_api.schemas import Pagination, Registries
+from offsets_db_api.schemas import Pagination, ProjectTypes, Registries
 from offsets_db_api.security import check_api_key
 from offsets_db_api.sql_helpers import apply_filters, apply_sorting, handle_pagination
 
 router = APIRouter()
 logger = get_logger()
 
 
+@router.get('/types', response_model=ProjectTypes, summary='Get project types')
+@cache(namespace=CACHE_NAMESPACE)
+async def get_project_types(
+    request: Request,
+    top_n: int = Query(6, description='Number of top project types to return'),
+    session: Session = Depends(get_session),
+    authorized_user: bool = Depends(check_api_key),
+):
+    """Get project types"""
+    logger.info(f'Getting project types: {request.url}')
+
+    statement = (
+        select(ProjectType.project_type, func.sum(Project.issued).label('total_issued'))
+        .join(Project, col(Project.project_id) == col(ProjectType.project_id))
+        .group_by(ProjectType.project_type)
+        .order_by(func.sum(Project.issued).desc())
+    )
+
+    result = session.exec(statement).all()
+    top = [project_type for project_type, _ in result[:top_n]]
+    others = [project_type for project_type, _ in result[top_n:]]
+    return ProjectTypes(top=top, others=others)
```

Comment:
Are we going to support the general `/projects?project_type=Other` query on the `/projects` endpoint? I think that would be helpful because including all of the "others" entries would make for a huge request URL.

If that's the case, should we use consistent naming here? 
```suggestion
    return ProjectTypes(Top=top, Other=others)
```

Reply:
one idea that comes to mind is doing something along these lines: 

update the` /projects` endpoint by adding the `top_project_types` parameter as a query parameter, and handle the `project_type=Other` logic when it's used in combination with `top_project_types`.


`GET /projects?project_type=Other&top_project_types=Type1,Type2,Type3` 

this would translate in using the `NOT IN` clause on on the backend to exclude the top project types if `project_type=Other` is detected.


```sql
SELECT ...
....
WHERE project_type NOT in top_project_types
```

one advantage of this approach is that we avoid having to query the database to get the list of `Other` types and instead take advantage of the fact that we already know which types are in `Top`

@katamartin, how does this sound?

Reply:
@andersy005 hm I'm not sure that pattern would be able to handle a case where a user tries to filter to see projects where `project_type` is either `REDD+` or `Other`. Do you see a way of making that work with what you suggest?

We don't necessarily need to handle querying by `Other` if you can confirm that, e.g., the search above could be accomplished by passing the full list of "other" `project_type`s and `REDD+`! It looks like the character limit for URLs fetched in JS [varies by browser](https://www.geeksforgeeks.org/maximum-length-of-a-url-in-different-browsers/), but is roughly 2000.

At a high-level, it seems like there should be a way to make the SQL query performant. Do we have an index setup on `project_type`? Should we consider caching the most common `project_type`s?